### PR TITLE
Resolves Warning SYSLIB021 - Derived cryptographic types are obsolete

### PR DESCRIPTION
**Pull request by Eneuman**
This PR solves the SYSLIB021 - Derived cryptographic types are obsolete warning.
It also introduces tests for the MacAddressHash functionallity.

The RunCommandAndGetOutput function was changed to use the existing `_platform.ExecuteAndReturnOutput` to align the way the program executes externa processes. The timeout was set to 30 sec to make sure the command have enought time to finish. The process will like finish within a second under normal conditions.

### CODE REVIEW

**Comment by Eneuman** [comment]
It would be great if we could also add a tests for Mono.
I can add it if someone can get the output from the "ifconfig -a" command.

File: src/common/Logging/MacAddressHash/MACInformationProvider.cs [hunk 1]
```diff
@@ -69,7 +68,7 @@ internal static class Mono
         /// Check if there is a persisted value otherwise calculates and persist a new one
         /// </summary>
         /// <returns>The hash of the mac address</returns>
-        private string GetMacAddressHash()
+        public string GetMacAddressHash()
```
Comment (hsubramanianaks) [inline src/common/Logging/MacAddressHash/MACInformationProvider.cs:71]: @Eneuman The reason for making this public is for UT's ?

**Review by hsubramanianaks — CHANGES_REQUESTED** [review]
@Eneuman Thanks for the changes, requested small changes.

File: src/common/Logging/MacAddressHash/MACInformationProvider.cs (hunk 1 above)
Comment (Eneuman) [reply src/common/Logging/MacAddressHash/MACInformationProvider.cs:71]: @hsubramanianaks Yeah, it's for making it public for UT. I tried using `protected` but it wasn't enought.

File: src/common/Logging/MacAddressHash/MACInformationProvider.cs [hunk 2]
```diff
@@ -157,34 +156,19 @@ private void PersistMacAddressHash(string hashedMacAddress)
...
-                };
-                process.Start();
-                process.BeginOutputReadLine();
-                process.BeginErrorReadLine();
-                process.WaitForExit();
-                process.Close();
-                return processOutput.ToString();
+                (var exitCode, var output) = this._platform.ExecuteAndReturnOutput(commandName, 
+                                                                              commandArgs, 
+                                                                              timeout: TimeSpan.FromSeconds(30),
```
Comment (hsubramanianaks) [inline src/common/Logging/MacAddressHash/MACInformationProvider.cs:165]: @Eneuman can you document or leave a comment why it was set to 30 seconds timeout please ? Thank you.

**Comment by hsubramanianaks** [comment]
> ifconfig -a

@Eneuman if you have WSL2 you can run this command ```ifconfig -a ``` , if not let me know I can share some samples.

File: src/common/Logging/MacAddressHash/MACInformationProvider.cs (hunk 2 above)
Comment (Eneuman) [reply src/common/Logging/MacAddressHash/MACInformationProvider.cs:165]: @hsubramanianaks I changed the description in the PR :) Do you want me to add a comment to the code aswell?

Comment (hsubramanianaks) [reply src/common/Logging/MacAddressHash/MACInformationProvider.cs:165]: Thank you, description should be fine.

**Comment by hsubramanianaks** [comment]
> It would be great if we could also add a tests for Mono. I can add it if someone can get the output from the "ifconfig -a" command.

here is the sample output for ```ifconfig -a``` command (I have edited with dummy values).

```bond0: flags=5122<BROADCAST,MASTER,MULTICAST>  mtu 1500
        ether e6:0e:ae:14:6c:da  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

dummy0: flags=130<BROADCAST,NOARP>  mtu 1500
        ether 6a:c7:29:b2:dc:9e  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1280
        inet 0.0.0.0  netmask 0.0.0.0  broadcast 0.0.0.0
        inet6 fe80::215:5dff:feb6:b81  prefixlen 64  scopeid 0x20<link>
        ether 00:15:5d:b6:0b:81  txqueuelen 1000  (Ethernet)
        RX packets 640092  bytes 605988996 (605.9 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 77211  bytes 5949128 (5.9 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 58608  bytes 361924804 (361.9 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 58608  bytes 361924804 (361.9 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

sit0: flags=128<NOARP>  mtu 1480
        sit  txqueuelen 1000  (IPv6-in-IPv4)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

tunl0: flags=128<NOARP>  mtu 1480
        tunnel   txqueuelen 1000  (IPIP Tunnel)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0 ````

File: src/common.tests/Logging/MacAddressHash/MacInformationProviderTests.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+using Microsoft.BridgeToKubernetes.TestHelpers;
+using System;
+using Xunit;
+
+namespace Microsoft.BridgeToKubernetes.Common.Tests.Logging.MacAddressHash
+{
+    public class MacInformationProviderTests : TestsBase
+    {
+        [Fact]
+        public void GetMacAddressHashOnWindows()
```
Comment (hsubramanianaks) [inline src/common.tests/Logging/MacAddressHash/MacInformationProviderTests.cs:14]: @Eneuman can you add one more test for Mac/Linux for mach address hash please ? Thank you. I have shared a sample output for ```ifconfig```, please check.

Comment (Eneuman) [reply src/common.tests/Logging/MacAddressHash/MacInformationProviderTests.cs:14]: @hsubramanianaks I have added a test for Linux :)

**Comment by Eneuman** [comment]
@hsubramanianaks I have another suggestion, why dont we just replace this code with .Net nativ plattform independent code like this?

```
var firstInterface = NetworkInterface.GetAllNetworkInterfaces().First();
var macAddressInBytes = firstInterface.GetPhysicalAddress().GetAddressBytes();
var hashedMacAddressByteArray = MD5.HashData(macAddressInBytes);
var hashedMacAddress= BitConverter.ToString(hashedMacAddressByteArray).Replace("-", string.Empty).ToLowerInvariant();

```

**Comment by hsubramanianaks** [comment]
> @hsubramanianaks I have another suggestion, why dont we just replace this code with .Net native plattform independent code like this?
> 
> ```
> var firstInterface = NetworkInterface.GetAllNetworkInterfaces().First();
> var macAddressInBytes = firstInterface.GetPhysicalAddress().GetAddressBytes();
> var hashedMacAddressByteArray = MD5.HashData(macAddressInBytes);
> var hashedMacAddress= BitConverter.ToString(hashedMacAddressByteArray).Replace("-", string.Empty).ToLowerInvariant();
> ```
> 
> I switched to MD5 because it's the fastest algorithm, and we are not doing encryption here ;)

yes if it make sense to do in this PR please go ahead.

**Comment by Eneuman** [comment]
@hsubramanianaks I'll make a seperate PR for this :) This PR can just solve the warning.

**Review by Tatsinnit — APPROVED** [review]
☕️🙏 LGTM - Hope fully we got his all tested et al from internal channels. thanks heaps @hsubramanianaks and @Eneuman 🎊

**Review by hsubramanianaks — APPROVED** [review]
testing team approved these changes.

**Pull request merged**